### PR TITLE
XHR#fetchText now returns Response

### PR DESCRIPTION
**Pull request by jridgewell**
Partial for #9448.

### CODE REVIEW

File: test/functional/test-batched-xhr.js [hunk 1]
```diff
@@ -213,10 +213,8 @@ describes.sandboxed('BatchedXhr', {}, () => {
       return Promise.all([
         xhr.fetchText('/get?k=v1', {method: 'POST', body: {}}),
         xhr.fetchText('/get?k=v1', {method: 'POST', body: {}}),
-      ]).then(results => {
+      ]).then(() => {
         expect(fetchStub).to.be.calledTwice;
-        expect(results[0]).to.equal(TEST_RESPONSE);
-        expect(results[1]).to.equal(TEST_RESPONSE);
```
Comment (cvializ) [inline test/functional/test-batched-xhr.js]: why not check that the values match `TEST_RESPONSE`?

Comment (jridgewell) [inline test/functional/test-batched-xhr.js]: Just a duplicate of the text above.

Comment (cvializ) [inline test/functional/test-batched-xhr.js]: But I notice you don't call `.then(res => res.text())` here, is that intentional? Is it not necessary for a POST? I believe checking the results matching the `TEST_RESPONSE` a second time for the POST method may catch unexpected behavior.

Comment (jridgewell) [inline test/functional/test-batched-xhr.js]: Processing the request body isn't necessary for the request to happen.

Comment (cvializ) [inline test/functional/test-batched-xhr.js]: ah makes sense

**Comment by jridgewell** [comment]
Approval? 😛

**Comment by cvializ** [comment]
LGTM

**Pull request merged**